Fixing some issues related to enumeration of COM ports

svhelper opened the pull request:
Hello,
Can you please apply this small fixes and rebuild .exe file?
- the first issue was fixed: the 2nd item will be destroyed during enumeration of usb-device names
- the second one related to complex usb-devices (composite devices, that contains few sub-devices and the MatchingDeviceId may have names starts from "usb\class_xx\")

svhelper committed:
Fixes the destroy of the 2nd item of COM-ports list

svhelper committed:
Added filling the list with available COM-ports, that without regular USB device name location (like usb\class_01\ etc)

Grovkillen commented:
I have compiled a v0.04.008 for you to test. If it does what you expect I'll merge and release your update. Thanks!

https://www.dropbox.com/t/riJcwlmh7VTAVJW6

svhelper commented:
> 
> 
> I have compiled a v0.04.008 for you to test. If it does what you expect I'll merge and release your update. Thanks!
> 
> https://www.dropbox.com/t/riJcwlmh7VTAVJW6

Sorry, probably you uploaded the previous version by mistake
![image](https://user-images.githubusercontent.com/22442675/135817155-c08749de-e6e7-4dd0-af28-ae7127af3b5f.png)

Grovkillen commented:
Ah! Sorry :)

Try this one: https://www.dropbox.com/s/prf6qlx369ij73x/ESP.Easy.Flasher.exe?dl=1



// Jimmy Westberg
0706505709
٩(- ̮̮̃-̃)۶

Den 2021-10-04 10:19:31 skrev svhelper ***@***.***>:
I have compiled a v0.04.008 for you to test. If it does what you expect I'll merge and release your update. Thanks!
https://www.dropbox.com/t/riJcwlmh7VTAVJW6 [https://www.dropbox.com/t/riJcwlmh7VTAVJW6]
Sorry, probably you uploaded the previous version by mistake
[image] [https://user-images.githubusercontent.com/22442675/135817155-c08749de-e6e7-4dd0-af28-ae7127af3b5f.png]
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub [https://github.com/Grovkillen/ESP_Easy_Flasher/pull/26#issuecomment-933254708], or unsubscribe [https://github.com/notifications/unsubscribe-auth/AGLYCYUVST4J6TZL2ZAH4UTUFFPRHANCNFSM5FHUGMHQ].
Triage notifications on the go with GitHub Mobile for iOS [https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&amp;mt=8&amp;pt=524675] or Android [https://play.google.com/store/apps/details?id=com.github.android&amp;referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub].

svhelper committed:
Fixed COM-port enumeration issues:
- the highest available in Windows COM255
- incorrect comparison of execution code result

svhelper commented:
Now the exe is correct, thanks a lot
But I found and fix 2 extra issues
I tested on Macro Scheduler 15 and all done fine
![image](https://user-images.githubusercontent.com/22442675/135827041-b6f0a5e8-0d66-4670-b089-8618820dae56.png)

just for comparison, here the previous list:
![image](https://user-images.githubusercontent.com/22442675/135827244-6763fc6a-ea6c-4a7d-afaf-269d63dbd7fc.png)

Any way, because the previous exe contains my mistakes (check of execution result code), the additional loop for generation and test of exe-file should be done.

Sorry for your time

Grovkillen commented:
No problem! If this file work as expected then please change the line 4 (version number to "Let>VERSION=0.04.009". After that I will merge and release the official way.

https://www.dropbox.com/s/prf6qlx369ij73x/ESP.Easy.Flasher.exe?dl=1

svhelper committed:
Release VERSION=0.04.009

svhelper commented:
Verified - All OK
I changed the version to `0.04.009` and commit as you required.

Grovkillen commented:
Thanks for the fix!
